### ISSUE TITLE
Testen mit Mockito

### ISSUE THREAD

**Issue by kkaiser12** [sweIhm-ws2018-19/skillproject-fr-14#31]
Da ich in diesem Sprint ansonsten relativ wenig Arbeit habe, hab ich mir jetzt mal angeschaut, wie bei der Gruppe mit 100% Testabedeckung das Testen funktioniert. Die nutzen eine Library namens Mockito, mit der man Objekte simulieren kann, die außerhalb des eigentlichen Javacodes erzeugt werden, also z.B. den Input, den Alexa bekommt. Ich hab auch schon mal ein kleines Beispiel gemacht (LaunchRequestHandlerTest) , das könnt ihr euch ja mal anschauen und dann sagen, ob ihr das gerne nutzen würdet oder nicht. Ich finde es  ziemlich gut, aber ich glaube, man muss sich schon erst etwas einarbeiten.

Ich weiß nicht genau, ob ihr die library über git bekommt, ansonsten könnt ihr sie hier downloaden:
[https://bintray.com/mockito/maven/mockito-development/2.23.9?versionPath=%2Fmockito%2Fmaven%2Fmockito-development%2F2.23.9#files/org%2Fmockito%2Fmockito-core%2F2.23.9](https://bintray.com/mockito/maven/mockito-development/2.23.9?versionPath=%2Fmockito%2Fmaven%2Fmockito-development%2F2.23.9#files/org%2Fmockito%2Fmockito-core%2F2.23.9)

(Bei Files unter dem Pfad /org/mockito/mockito-core/2.23.9 die Datei mockito-core-2.23.9.jar auswählen.)

**Comment by ngaeng** [comment]
Vielen Dank! Ich würde sagen wir machen das auf jeden Fall so.

**Comment by Chrysotomus** [comment]
Jo, danke dir. Ich stimme auch zu. Ich werde damit auch noch unsere Calendar (zukünftig Day)-Klasse testen.

**Comment by ngaeng** [comment]
Im aktuellen commit sind die tests für den Launch Handler auskommentiert, weil maven sonst beim build  leider einen error wirft.

**Comment by ngaeng** [comment]
![image](https://user-images.githubusercontent.com/43876351/50590114-dea25600-0e89-11e9-9fd9-88120d360158.png)

Das sind die Klassen mit noch ungetestetem Code. Der Launch Request Handler wäre nicht so leicht zu covern, wegen der Datenbank. Am meisten Sinn macht wahrscheinlich die Day Klasse weil wir damit ~15% gutmachen könnten.

**Comment by Chrysotomus** [comment]
Ehrlich gesagt habe ich diese Ferien noch gar nichts für das Projekt oder im Allgemeinen für die Uni gemacht. Ich bin seit Beginn der Ferien krank und immer noch nicht genesen xD
Jetzt müsste ich allerdings genau wissen, wie genau wir uns das mit der Weiterentwicklung vorstellen. Wie soll meine Day Klasse die Accountdaten holen? Soll das über die Datenbank gelöst werden? Eine möglichst schnelle Antwort wäre super, da die Zeit relativ knapp wird, sobald ich erstmal gesund bin.....
@kkaiser12 @ngaeng 

PS: Ich versuche danach noch die Tests zu schreiben. Ich kann aber nichts versprechen, weil ich noch recht viel für Netzwerke vorbereiten muss und praktisch noch keine Prüfungsvorbereitung für die anderen Fächer hatte

**Comment by ngaeng** [comment]
Gute Besserung auf jeden Fall!

Wir könnten das auf jeden Fall mit der Datenbank lösen. Man müsste im LaunchRequestHandler in der Methode wo die Fächer in die Datenbank gespeichert werden zwei weitere key-value Paare für Username und Passwort hinzufügen. Diese kann man dann folgendermaßen an anderer Stelle abfragen:

```Java
AttributesManager attributesManager = input.getAttributesManager();
Map<String, Object> persistentAttributes = attributesManager.getPersistentAttributes();
Object username = persistentAttributes.get("Username");
```
Falls das mit den Tests noch geht wärs super, ansonsten fehlen uns halt die paar Punkte, sollte jetzt denk ich mal auch nicht der Ausmacher sein wenns nicht ganz desaströs läuft, hauptsache ist jetzt erstmal, dass du schnell wieder gesund wirst

Edit: Ah ja wenn man in der Klasse noch nen kleineren Teil einfach testen kann wär das auch schon gut, uns fehlen noch 6% zu 50%

**Comment by ngaeng** [comment]
Wir sind jetzt bei 53,8%, also alles easy :D
